server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java

```diff
@@ -81,8 +81,10 @@ public void testThreadNames() throws Exception {
             // or the ones that are occasionally come up from ESSingleNodeTestCase
             if (threadName.contains("[node_s_0]") // TODO: this can't possibly be right! single node and integ test are unrelated!
                 || threadName.contains("Keep-Alive-Timer")
+                || threadName.contains("readiness-service")
                 || threadName.contains("JVMCI-native") // GraalVM Compiler Thread
-                || threadName.contains("readiness-service")) {
+                || threadName.contains("file-settings-watcher")
+                || threadName.contains("FileSystemWatchService")) {
```

Comment:
I agree that's this is tedious to modify each time. We need to explicitly add each thread that's not in any threadpool. When I worked on the readiness-service, I was advised not to use the thread pools for threads that will constantly be running, so I followed the same approach here. `FileSystemWatchService` comes from the JDK, so there's no avoiding adding that one.

Is it OK if I followed up with a new PR to tackle this problem, rather than extend this PR further? I think I should be able to ensure all standalone Elasticsearch threads are registered in a way that this list doesn't have to be manually modified.

Reply:
A followup is great